akka-http-core/src/main/scala/akka/http/scaladsl/model/ws/UpgradeToWebSocket.scala

```diff
@@ -54,6 +57,31 @@ trait UpgradeToWebSocket extends jm.ws.UpgradeToWebSocket {
     subprotocol: Option[String]                   = None): HttpResponse =
     handleMessages(scaladsl.Flow.fromSinkAndSource(inSink, outSource), subprotocol)
 
+  def handleStrictMessages(
+    handlerFlow: Graph[FlowShape[StrictMessage, Message], Any],
+    subprotocol: Option[String]                                = None)(implicit mat: Materializer, timeout: FiniteDuration): HttpResponse =
+    handleMessages(scaladsl.Flow[Message].mapAsync(1) {
+      case streamed: TextMessage.Streamed   ⇒ streamed.toStrict(timeout)
+      case streamed: BinaryMessage.Streamed ⇒ streamed.toStrict(timeout)
```

Comment:
I think we also need a max-bytes limit in addition to the timeout here, with a fast client I could potentially OOM the server by just streaming a huge amount of streamed data. On the entity side this is protected by the `akka.http.parsing.max-content-length` so maybe we need either to use that or a corresponding new setting. Thoughts @raboof @jrudolph ?